src/app/services/note.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable } from '@angular/core';
+import { Note } from '../models/note';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable, catchError, of, tap } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NoteService {
+  httpOptions = {
+    headers: new HttpHeaders({'Content-Type': 'application/json'})
+  };
+  
+  API_URL: string = environment.apiUrl;
+  noteCategories: string[] = ['Work', 'School', 'Personal', ];
+  currentNote?: Note;
+  noteList!: Note[];
+
+  constructor(private http: HttpClient) {
+    this.getNotes();
```

Comment:
Generally we don't want to call anything in the constructor. This should be called in the component/page component you want to load the data in